experimental/packages/opentelemetry-instrumentation-http/test/functionals/http-enable.test.ts

```diff
@@ -908,4 +908,87 @@ describe('HttpInstrumentation', () => {
       });
     });
   });
+
+  describe('capturing headers as span attributes', () => {
+    beforeEach(() => {
+      memoryExporter.reset();
+    });
+
+    before(() => {
+      instrumentation.setConfig({
+        headersToSpanAttributes: {
+          client: { requestHeaders: ['X-Client-Header1'], responseHeaders: ['X-Server-Header1'] },
+          server: { requestHeaders: ['X-Client-Header2'], responseHeaders: ['X-Server-Header2'] },
+        }
+      });
+      instrumentation.enable();
+      server = http.createServer((request, response) => {
+        response.setHeader('X-Server-Header1', 'server123');
+        response.setHeader('X-Server-Header2', '123server');
+        response.end('Test Server Response');
+      });
+
+      server.listen(serverPort);
+    });
+
+    after(() => {
+      server.close();
+      instrumentation.disable();
+    });
+
+    it('should convert headers to span attributes', async () => {
+      await httpRequest.get(
+        `${protocol}://${hostname}:${serverPort}${pathname}`,
+        {
+          headers: {
+            'X-Client-Header1': 'client123',
+            'X-Client-Header2': '123client',
```

Comment:
```suggestion
            'X-ClienT-hEader2': '123client',
```
as above to verify that casing doesn't matter